src/js/actions/layers.js

```diff
@@ -482,6 +482,13 @@ define(function (require, exports) {
             return Promise.resolve();
         }
 
+        var upToDateDocument = this.flux.store("document").getDocument(document.id);
```

Comment:
@mcilroyc @volfied: I agree that there is a problem with our pattern of passing in an immutable model in actions. We should talk at the upcoming team meeting about the best possible replacement pattern.